src/charm.py

```diff
@@ -413,6 +422,9 @@ def _prometheus_layer(self) -> Layer:
                     "summary": "prometheus daemon",
                     "command": self._generate_command(),
                     "startup": "enabled",
+                    "environment": {
+                        "OTEL_RESOURCE_ATTRIBUTES": f"juju_application={self._topology.application},juju_model={self._topology.model},juju_model_uuid={self._topology.model_uuid},juju_unit={self._topology.unit},juju_charm={self._topology.charm_name}"
```

Comment:
Same comment from @PietroPasotti as here: https://github.com/canonical/tempo-worker-k8s-operator/pull/47#discussion_r1839849932

probably applies in prometheus as well. Looking into it.